index.js

```diff
@@ -133,17 +145,29 @@ module.exports = function (log, indexesPath) {
     saveBitsetFile(filename, index.seq, index.bitset, cb)
   }
 
-  function growCoreIndex(coreIndex, Type) {
-    debug('growing index')
-    const newArray = new Type(coreIndex.tarr.length * 2)
-    newArray.set(coreIndex.tarr)
-    coreIndex.tarr = newArray
+  function savePrefixIndex(name, prefixIndex, count, cb) {
+    debug('saving prefix index: %s', name)
+    const num = prefixIndex.prefix
+    const filename = path.join(indexesPath, name + `.${num}prefix`)
+    saveTypedArrayFile(filename, prefixIndex.seq, count, prefixIndex.tarr, cb)
+  }
+
+  function growTarrIndex(index, Type) {
+    let newLength = index.tarr.length * 2
+    // FIXME: even better would be to bring back the exponential jumps, but
```

Comment:
Yes, we really need to exponential back here. There is quite a big perf diff. Maybe with a trim at save function.

Reply:
So, I'm not sure what to do here, but I'll share some numbers:

- exponential jumps unlimited
  - `and(blog, or(author1, author2))` with every index missing: 4400ms
  - `and(blog, or(author1, author2))` with non-core indexes missing: 3000ms
  - `and(blog, or(author1, author2))` warm query: 18ms
  - `equal(seekVoteLink, msgid, prefix)` with every index missing: 3400ms
  - `equal(seekVoteLink, msgid, prefix)` with non-core indexes missing: 2000ms
  - `equal(seekVoteLink, msgid, prefix)` warm query: 45ms
- exponential jumps limited to 64000
  - `and(blog, or(author1, author2))` with every index missing: 4600ms
  - `and(blog, or(author1, author2))` with non-core indexes missing: 3000ms
  - `and(blog, or(author1, author2))` warm query: 18ms
  - `equal(seekVoteLink, msgid, prefix)` with every index missing: 3500ms
  - `equal(seekVoteLink, msgid, prefix)` with non-core indexes missing: 2100ms
  - `equal(seekVoteLink, msgid, prefix)` warm query: 40ms

So there's a small CPU perf difference, but more significant storage difference.

Reply:
So I think I'll put unlimited exponential jumps back, and for the storage solution, compression. What do u think?

Reply:
There is a count in saveCoreIndex that could be used to just trim of the 0s at the end I think on save. As for compression maybe we can do that in a separate PR?